### PR TITLE
feat: added geoip stage to loki.process

### PR DESCRIPTION
**Pull request by akselleirv**
#### PR Description

Port of the [geoip stage](https://grafana.com/docs/loki/latest/clients/promtail/stages/geoip/#geoip) in Promtail.

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md
Comment (rfratto) [inline CHANGELOG.md:16]:
```suggestion
 - Support `stage.geoip` in `loki.process`. (@akselleirv)
```

File: docs/sources/flow/reference/components/loki.process.md
Comment (rfratto) [inline docs/sources/flow/reference/components/loki.process.md:1367]:
```suggestion
The `stage.geoip` inner block configures a processing stage that reads an ip address and populates the labelset with geoip fields. Maxmind’s GeoIP2 databse is used for the lookup.
```

File: go.mod
Comment (rfratto) [inline go.mod:210]: Can you add these to the existing require block above and re-run `go mod tidy`? I'd like to keep the number of require blocks to a minimum.

File: docs/sources/flow/reference/components/loki.process.md
Comment (rfratto) [inline docs/sources/flow/reference/components/loki.process.md:1385]: We typically put detail like this following the arguments table for the block so the table is close as it can be to the header, can you move it down below?

**Review by rfratto — APPROVED** [review]
LGTM, waiting for content review from @clayton-cornell before merging :) 

Thanks for the contribution!

File: docs/sources/flow/reference/components/loki.process.md
Comment (akselleirv) [reply docs/sources/flow/reference/components/loki.process.md]: The added labels are already well documented in the two examples, so I decided to remove it from the introduction.

File: docs/sources/flow/reference/components/loki.process.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:1367]:
```suggestion
The `stage.geoip` inner block configures a processing stage that reads an IP address and populates the labelset with geoip fields. Maxmind’s GeoIP2 database is used for the lookup.
```

File: docs/sources/flow/reference/components/loki.process.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:1410]:
```suggestion
If only a subset of these labels is required, you can chain the above pipeline with the `label_drop` or `label_keep` stage.
```

File: docs/sources/flow/reference/components/loki.process.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:1430]:
```suggestion
Only the labels listed in the `values` list in the `label_keep` stage are sent to Loki.
```

File: docs/sources/flow/reference/components/loki.process.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:1451]:
```suggestion
All the labels except the ones listed under `label_drop` stage are sent to Loki.
```

**Review by clayton-cornell — APPROVED** [review]
Some minor change/suggestions

**Comment by akselleirv** [comment]
Thank you @clayton-cornell! I’ve updated the PR with your suggestions.

**Comment by clayton-cornell** [comment]
@rfratto Docs look good to me now. Over to you.

**Pull request merged**
